docs/proxy/README.md

```diff
@@ -0,0 +1,133 @@
+* Additional automation will be required to identify the appropriate IP and set it for listen_address and broadcast_address
+* Only one Cassandra pod may be started per worker, regardless of `allowMultiplePodsPerWorker` setting.
+
+### Host Port
+
+Host port is similar to host network, but instead of being applied at the pod level, it is applied to specified containers within the pod. For each port listed in the container's block a hostPort: external_port key value is included. external_port is the port number on the Kubernetes worker that should be forwarded to this container's port.
+
+At this time we do not allow for modifying the cassandra container via podTemplateSpec, thus configuring this value is not possible without patching each rack's stateful set.
+
+**Pros**
+
+* External connectivity is possible as the service is available at the nodes IP instead of an IP internal to the Kubernetes cluster.
+* Easier configuration a separate container to determine the appropriate IP is no longer required.
+
+**Cons**
+
+* If a pod is rescheduled the IP address of the pod can change
+* In some K8s distributions this is a privileged operation
+* Only one Cassandra pod may be started per worker, regardless of allowMultiplePodsPerWorker setting.
+* Not recommended according to K8s [Configuration Best Practices](https://kubernetes.io/docs/concepts/configuration/overview/#services).
+
+## Services
+
+If the application is running within the same Kubernetes cluster as the Cassandra cluster connectivity is simple. cass-operator exposes a number of services representing a Cassandra cluster, datacenters, and seeds. Applications running within the same Kubernetes cluster may leverage these services to discover and identify pods within the target C* cluster.
+
+External applications do not have access to this information via DNS as internal applications do. It is possible to forward DNS requests to Kubernetes from outside the cluster and resolve configured services. Unfortunately, this will provide the internal pod IP addresses and not those routable unless Network Supported Direct Access is possible within the environment. In most scenarios, external applications will not be able to leverage the exposed services from cass-operator.
+
+### Load Balancer
+
+It is possible to configure a service within Kubernetes outside of those provided by cass-operator that is accessible from outside of the Kubernetes cluster. These services have a type: LoadBalancer key in the spec block. In most cloud environments this results in a native cloud load balancer being provisioned to point at the appropriate pods with an external IP. Once the load balancer is provisioned running kubectl get svc will display the external IP address that is pointed at the C* nodes.
+
+**Pros**
+
+* Available from outside of the cluster
+
+**Cons**
+
+* Requires WhiteListPolicy Load Balancing Policy (LBP) to restrict attempts by the drivers to connect directly with pods
```

Comment:
I don't see an example of this. Might be nice in the future as this looks like a relatively straightforward way of just getting _something_ working.

Reply:
Heh, this needs to change a little as we removed the `WhiteListPolicy`. Now it's a simple implementation of `AddressTranslator` (see our [docs](https://docs.datastax.com/en/developer/java-driver/4.7/manual/core/address_resolution/#driver-side-address-translation) and my [example](https://github.com/datastax/cass-operator/blob/proxy-config/docs/proxy/sample-java-application/src/main/java/com/datastax/kubernetes/KubernetesIngressAddressTranslator.java).